proxy.ts

```diff
@@ -40,4 +23,43 @@ export function getProxyUrl(reqUrl: url.Url): url.Url {
     }
 
     return proxyUrl;
+}
+
+
+export function checkBypass(reqUrl: url.Url): boolean {
+    if (!reqUrl.hostname) {
+        return false
+    }
+
+    let noProxy = process.env["no_proxy"] || process.env["NO_PROXY"] || '';
+    if (!noProxy) {
+        return false
+    }
+
+    // Determine the request port
+    let reqPort: number
+    if (reqUrl.port) {
```

Comment:
Url.port is empty for urls like `http://foo.com` so i have a few if blocks to set based on the default port